crates/oxc_linter/src/rules/oxc/ifs_same_conditions.rs

```diff
@@ -0,0 +1,113 @@
+use itertools::Itertools;
+use oxc_ast::{
+    ast::{Expression, IfStatement, Statement},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+use oxc_span::cmp::ContentEq;
+
+fn ifs_same_conditions_diagnostic(span1: Span, span2: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Unexpected consecutive `if` statements with the same condition.")
+        .with_help(
+            "The second branch can never execute. Consider removing it or changing the condition.",
+        )
+        .with_labels([span1, span2])
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct IfsSameConditions;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Checks for consecutive `if` statements with the same condition.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// This is probably a mistake, as the second `if` statement will never be executed.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// if (a) {
+    /// } else if (a) {
+    /// }
+    ///
+    /// if (a) {
+    /// } else if (b) {
+    /// } else if (a) {
+    /// }
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// if (a) {
+    /// } else if (b) {
+    /// }
+    ///
+    /// if (a) {
+    /// } else if (b) {
+    /// } else {
+    /// }
+    /// ```
+    IfsSameConditions,
+    correctness
+);
+
+impl Rule for IfsSameConditions {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::IfStatement(if_stmt) = node.kind() else {
+            return;
+        };
+
+        let tests = collect_if_tests(if_stmt);
+        if tests.len() < 2 {
+            return;
+        }
+
+        tests.iter().tuple_combinations().for_each(|(first_test, second_test)| {
+            if first_test.content_eq(second_test) {
```

Comment:
`content_eq` doesn't consider `(foo as string)` and `(foo)` the same?
this means that we can't catch cases like `if (foo) {} else if (foo as string) {}`
should `content_eq` use https://github.com/oxc-project/oxc/blob/28da77195b88c672f2260dcbf9773a6b8f8c347d/crates/oxc_ast/src/ast_impl/js.rs#L208-L222 and ignore these?

Regardless, doesn't make sense to change in this PR, just something to consider